src/pages/dataElements/List.tsx

```diff
@@ -26,35 +24,35 @@ type FilteredDataElement = Pick<DataElement, (typeof filterFields)[number]>
 
 type DataElements = GistCollectionResponse<FilteredDataElement>
 
-const defaulHeaderColumns: SelectedColumns<FilteredDataElement> = [
-    {
-        modelPropertyName: 'name',
-        label: i18n.t('Name'),
-    },
-    { modelPropertyName: 'domainType', label: i18n.t('Domain') },
-    { modelPropertyName: 'valueType', label: i18n.t('Value') },
-    { modelPropertyName: 'lastUpdated', label: i18n.t('Last updated') },
-    { modelPropertyName: 'sharing', label: i18n.t('Public access') },
-]
-
 export const Component = () => {
+    const { columns, query: listViewQuery } = useModelListView()
     const initialParams = useQueryParamsForModelGist()
     const { refetch, error, data } = useModelGist<DataElements>(
         'dataElements/gist',
         {
             fields: filterFields.concat(),
             ...initialParams,
         },
-        // refetched on mount by useSectionListParamsRefetch below
+        // refetched on mount by effect below
         { lazy: true }
     )
-
-    useSectionListParamsRefetch(refetch)
+    useEffect(() => {
+        // wait to fetch until selected-columns are loaded
+        // so we dont fetch data multiple times
+        if (listViewQuery.isLoading) {
+            return
+        }
+        refetch({
+            ...initialParams,
+            fields: columns
+                .map((column) => getFieldFilterFromPath(column.path, 0))
+                .concat('id'),
+        })
+    }, [refetch, initialParams, columns, listViewQuery.isLoading])
```

Comment:
I don't think the loading state of the query is an implementation detail, but I agree that could be a different solution, especially with how our usage of `refetch` works. However, we will probably move the list-query to use a  reactive-query instead of refetch to deal with updating the list. And then we would just create the query in render, and set `lazy` or `enabled: listViewQuery.isLoading`. 
Generally I want less imperative code for dealing with queries, and a callback would just encourage that.

Reply:
> However, we will probably move the list-query to use a reactive-query instead of refetch to deal with updating the list. And then we would just create the query in render, and set lazy or enabled: listViewQuery.isLoading (depending on how we align useDataQuery with react-query.

:ok_hand: